File: app/src/main/java/com/nimtego/plectrum/presentation/information_view/song/SongInformationFragment.java
package com.nimtego.plectrum.presentation.information_view.song;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.nimtego.plectrum.R;
import com.nimtego.plectrum.presentation.base.BaseFragment;
import com.nimtego.plectrum.presentation.information_view.song.model.SongDetailsModel;
import com.nimtego.plectrum.presentation.information_view.song.model.SongDetailsModelK;
import com.squareup.picasso.Callback;
import com.squareup.picasso.Picasso;

import static com.nimtego.plectrum.presentation.utils.IpTags.SONG_ID;

@Deprecated
public class SongInformationFragment
        extends BaseFragment<SongInformationContract.Presenter>
        implements SongInformationContract.View<SongInformationContract.Presenter> {

    private ImageView songImage;
    private TextView information;

    public static SongInformationContract.View newInstance(final String content) {
        final SongInformationContract.View fragment = new SongInformationFragment();
        final Bundle arguments = new Bundle();
        arguments.putString(SONG_ID.name(), content);
        ((BaseFragment) fragment).setArguments(arguments);
        return fragment;
    }

    @Override
    public SongInformationContract.Presenter supplyPresenter() {
        return new SongInformationPresenter();
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.information_song_form, container, false);
        information = view.findViewById(R.id.author);
        songImage = view.findViewById(R.id.song_image);

        mPresenter.viewReady(getArguments().getString(SONG_ID.name()));
        return view;
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        /* this.mPresenter.viewReady(getArguments().getString(ALBUM_ID.name()));*/
    }

    @Override
    public void render(SongDetailsModelK songDetailsModel) {
        StringBuilder sb = new StringBuilder();
        sb.append(songDetailsModel.getSongName()).append("\n\n")
                .append(songDetailsModel.getSongAlbumName()).append("\n\n")
                .append(songDetailsModel.getSongArtistName()).append("\n\n")
                .append(songDetailsModel.getReleaseDate()).append("\n\n")
                .append(songDetailsModel.getSongPrice()).append("\n\n");
        information.setText(sb);
        Picasso.get().load(songDetailsModel.getSongArtwork()
                .replace("100x100", "400x400"))
                .into(songImage, new Callback() {
                    @Override
                    public void onSuccess() {
                  /*      if (pb != null)
                            pb.setVisibility(View.GONE);*/
                    }

                    @Override
                    public void onError(Exception e) {
                   /*     if (pb != null)
                            pb.setVisibility(View.GONE);*/
                    }
                });
    }
}

File: app/src/main/java/com/nimtego/plectrum/presentation/old/information_view/song/SongInformationFragment.kt
package com.nimtego.plectrum.presentation.old.information_view.song

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import com.arellomobile.mvp.presenter.InjectPresenter
import com.arellomobile.mvp.presenter.ProvidePresenter
import com.nimtego.plectrum.R
import com.nimtego.plectrum.presentation.base.BaseFragment
import com.nimtego.plectrum.presentation.old.information_view.song.model.SongDetailsModel
import com.nimtego.plectrum.presentation.utils.IpTagsK
import com.squareup.picasso.Callback
import com.squareup.picasso.Picasso

class SongInformationFragment : BaseFragment(), SongInformationView  {
    private var songImage: ImageView? = null
    private var information: TextView? = null

    @InjectPresenter
    internal lateinit var presenter: SongInformationPresenterImpl

    @ProvidePresenter
    fun provideRepositoryPresenter(): SongInformationPresenterImpl {
        return SongInformationPresenterImpl()
    }
companion object {
    fun newInstance(content: String): SongInformationFragment {
        val fragment = SongInformationFragment()
        val arguments = Bundle()
        arguments.putString(IpTagsK.SONG_ID.name, content)
        fragment.arguments = arguments
        return fragment
    }
}

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater!!.inflate(R.layout.song_information_fragment, container, false)
        information = view.findViewById(R.id.author)
        songImage = view.findViewById(R.id.song_image)

        presenter.viewReady(arguments.getString(IpTagsK.SONG_ID.name))
        return view
    }

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        /* this.mPresenter.viewReady(getArguments().getString(ALBUM_ID.name()));*/
    }

    override fun render(songDetailsModel: SongDetailsModel) {
        val sb = StringBuilder()
        sb.append(songDetailsModel.songName).append("\n\n")
                .append(songDetailsModel.songAlbumName).append("\n\n")
                .append(songDetailsModel.songArtistName).append("\n\n")
                .append(songDetailsModel.releaseDate).append("\n\n")
                .append(songDetailsModel.songPrice).append("\n\n")
        information!!.text = sb
        Picasso.get().load(songDetailsModel.songArtwork!!
                .replace("100x100", "400x400"))
                .into(songImage!!, object : Callback {
                    override fun onSuccess() {
                        /*      if (pb != null)
                            pb.setVisibility(View.GONE);*/
                    }

                    override fun onError(e: Exception) {
                        /*     if (pb != null)
                            pb.setVisibility(View.GONE);*/
                    }
                })
    }
}